sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala

```diff
@@ -278,8 +278,16 @@ private[hive] object SparkSQLCLIDriver extends Logging {
 
     var ret = 0
     var prefix = ""
-    val currentDB = ReflectionUtils.invokeStatic(classOf[CliDriver], "getFormattedDb",
```

Comment:
the old code itself looks fine, why `CliDriver.getFormattedDb` returns wrong result?

Reply:
Yes, it will call SessionState.get().getCurrentDatabase and always return `default`

Reply:
it's not spark's `SessionState`, right? For spark cli, Spark's own `SessionState` is the source of truth. Are there other places trying to get state from Hive `SessionState`?

Reply:
Yes, it's hive's sessionState. 

> Are there other places trying to get state from Hive SessionState?

The hive‘s sessionState created here is used for option parsing and will be used in `org.apache.spark.sql.hive.client.HiveClientImpl`, which is OK.